tests_func/helpers/configuration.go

```diff
@@ -0,0 +1,67 @@
+package helpers
+
+import (
+	"bufio"
+	"fmt"
+	c "github.com/wal-g/wal-g/tests_func/config"
+	u "github.com/wal-g/wal-g/tests_func/utils"
+	"math/rand"
+	"os"
+	"strconv"
+)
+
+type UserConfiguration struct {
+	Username string
+	Password string
+	Dbname   string
+	Roles    []string
+}
+
+func GetConfiguration(testContext *TestContextType) []string {
+	portFactor := u.GetVarFromEnvList(testContext.Env, "TEST_ID")
+	netName := fmt.Sprintf("test_net_%s", portFactor)
+	dynamicConfig := getDynamicConfiguration(testContext)
+
+	curEnv := map[string]string{
+		"DOCKER_BRIDGE_ID": strconv.Itoa(rand.Intn(65535)),
+		"PORT_FACTOR":      portFactor,
+		"NETWORK_NAME":     netName,
+	}
+
+	return u.MergeEnvs(dynamicConfig, u.MapEnvToListEnv(curEnv))
+}
+
+func getDynamicConfiguration(testContext *TestContextType) []string {
+	curEnv := map[string]string{
+		"S3_HOST":              fmt.Sprintf("%s.%s", u.GetVarFromEnvList(testContext.Env, "S3_WORKER"), u.GetVarFromEnvList(testContext.Env, "TEST_ID")),
+		"S3_FAKE_HOST_PORT":    fmt.Sprintf("%s:%s", u.GetVarFromEnvList(testContext.Env, "S3_FAKE_HOST"), u.GetVarFromEnvList(testContext.Env, "S3_FAKE_PORT")),
+		"S3_ENDPOINT":          fmt.Sprintf("http://%s:%s", u.GetVarFromEnvList(testContext.Env, "S3_FAKE_HOST"), u.GetVarFromEnvList(testContext.Env, "S3_FAKE_PORT")),
+		"S3_ACCESS_SECRET_KEY": u.GetVarFromEnvList(testContext.Env, "MINIO_SECRET_KEY"),
+		"S3_ACCESS_KEY_ID":     u.GetVarFromEnvList(testContext.Env, "MINIO_ACCESS_KEY"),
+	}
+
+	return u.MapEnvToListEnv(curEnv)
+}
+
+func GetTestEnv(testContext *TestContextType) []string {
+	if testContext.Env != nil {
+		return testContext.Env
+	}
+	env := make([]string, 0)
+	envFile := c.Env["ENV_FILE"]
+	file, err := os.OpenFile(envFile, os.O_RDONLY, 0644)
+	if err != nil {
+		panic(err)
```

Comment:
You can handle error here gracefully. Return error from your function explicitly, and describe what's wrong in error message